resources/src/nextpvr/epg.py

```diff
@@ -840,35 +843,28 @@ def onAction(self, action):
             return
 
         self.ready = False
+        
         focusID = xbmcgui.WindowXML.getFocusId(self)
-        if focusID == scrollDownButtonId or focusID == scrollUpButtonId or focusID == numButtonId:
-            if actionID in MOVEMENT_LEFT:
-                if focusID == scrollDownButtonId:
-                    xbmcgui.WindowXML.setFocus(self, self.numButton)
-                if focusID == numButtonId:
-                    xbmcgui.WindowXML.setFocus(self, self.upArrow)
-            if actionID in MOVEMENT_RIGHT:
-                if focusID == scrollUpButtonId:
-                    xbmcgui.WindowXML.setFocus(self, self.numButton)
-                if focusID == numButtonId:
-                    xbmcgui.WindowXML.setFocus(self, self.downArrow)
-                if focusID == scrollDownButtonId:
-                    self.reFocus()
-        else:
+        if (focusID == passButtonID):
```

Comment:
Doesn't this work?

if focusID == scrollUpButtonId and actionID in MOVEMENT_LEFT:
   self. self.reFocus()

Reply:
That will self.reFocus when moving TO the scrollUpButton with a MOVEMENT_LEFT, which is probably not what you want.

Reply:
It shouldn't because the action relates to the control that has focus, but I can't build your code because of the unresolved conflicts to test myself.  Did you try it?

Reply:
Yes, I test everything. I wrote it originally as you describe, which seemed the most reasonable way for it work, but it doesn't work that way.